src/plugins/git/createGraph.js

```diff
@@ -123,34 +165,50 @@ class GitGraphCreator {
         payload: {name},
       };
       result.addNode(entryNode).addEdge(entryEdge);
-      if (entry.type === "commit") {
-        // We don't represent subproject commits in the graph.
-      } else {
-        let contentsNodeType;
-        if (entry.type === "tree") {
-          contentsNodeType = TREE_NODE_TYPE;
-        } else if (entry.type === "blob") {
-          contentsNodeType = BLOB_NODE_TYPE;
+      const contentsNodes: Node<NodePayload>[] = (() => {
+        if (entry.type !== "commit") {
+          let contentsNodeType;
+          if (entry.type === "tree") {
+            contentsNodeType = TREE_NODE_TYPE;
+          } else if (entry.type === "blob") {
+            contentsNodeType = BLOB_NODE_TYPE;
+          } else {
+            // eslint-disable-next-line no-unused-expressions
+            (entry.type: empty);
+            throw new Error(`Unknown entry type: ${entry.type}`);
+          }
+          const contentsNode: Node<TreeNodePayload> | Node<BlobNodePayload> = {
+            address: this.makeAddress(contentsNodeType, entry.hash),
+            payload: (({}: any): {||}),
+          };
+          return [contentsNode];
         } else {
-          // eslint-disable-next-line no-unused-expressions
-          (entry.type: empty);
-          throw new Error(`Unknown entry type: ${entry.type}`);
+          // One entry for each possible URL.
+          const urls = treeAndNameToSubmoduleUrls[tree.hash][name];
+          return urls.map((url): Node<SubmoduleCommitPayload> => ({
+            address: this.makeAddress(
+              SUBMODULE_COMMIT_NODE_TYPE,
+              submoduleCommitId(entry.hash, url)
+            ),
+            payload: {
+              hash: entry.hash,
+              url,
+            },
+          }));
         }
-        const contentsNode = {
-          address: this.makeAddress(contentsNodeType, entry.hash),
-          payload: {},
-        };
-        const contentsEdge = {
+      })();
+      contentsNodes.forEach((contentsNode) => {
+        const contentsEdge: Edge<HasContentsEdgePayload> = {
           address: this.makeAddress(
             HAS_CONTENTS_EDGE_TYPE,
             edgeID(entryNode.address, contentsNode.address)
           ),
           src: entryNode.address,
           dst: contentsNode.address,
-          payload: {},
+          payload: (({}: any): {||}),
```

Comment:
What does this mean?

Reply:
Yeah (\*sigh\*). This is to work around a long-standing bug in Flow: https://github.com/facebook/flow/issues/2977

Reply:
(The change is necessary in this commit because I added an explicit type to `contentsEdge` on the preceding line 201.)